diff --git a/pint/__init__.py b/pint/__init__.py
index a954828..ca61f13 100644
--- a/pint/__init__.py
+++ b/pint/__init__.py
@@ -15,7 +15,6 @@ from __future__ import annotations
 
 from importlib.metadata import version
 
-from .delegates.formatter._format_helpers import formatter
 from .errors import (  # noqa: F401
     DefinitionSyntaxError,
     DimensionalityError,
@@ -45,7 +44,6 @@ except Exception:  # pragma: no cover
     # so the reported version will be unknown
     __version__ = "unknown"
 
-
 #: A Registry with the default units and constants.
 _DEFAULT_REGISTRY = LazyRegistry()
 
@@ -55,46 +53,26 @@ application_registry = ApplicationRegistry(_DEFAULT_REGISTRY)
 
 def _unpickle(cls, *args):
     """Rebuild object upon unpickling.
-    All units must exist in the application registry.
 
     Parameters
     ----------
-    cls : Quantity, Magnitude, or Unit
-    *args
+    cls : Quantity, Unit, or Measurement
+    *args : tuple
+        Arguments to pass to the cls constructor.
+        All units must exist in the application registry.
 
     Returns
     -------
     object of type cls
-
     """
-    from pint.util import UnitsContainer
-
-    for arg in args:
-        # Prefixed units are defined within the registry
-        # on parsing (which does not happen here).
-        # We need to make sure that this happens before using.
-        if isinstance(arg, UnitsContainer):
-            for name in arg:
-                application_registry.parse_units(name)
+    # Prefixed units are defined within the registry
+    # on parsing (which does not happen here).
+    # We need to make sure that this happens before using.
+    application_registry.get()
 
     return cls(*args)
 
 
-def _unpickle_quantity(cls, *args):
-    """Rebuild quantity upon unpickling using the application registry."""
-    return _unpickle(application_registry.Quantity, *args)
-
-
-def _unpickle_unit(cls, *args):
-    """Rebuild unit upon unpickling using the application registry."""
-    return _unpickle(application_registry.Unit, *args)
-
-
-def _unpickle_measurement(cls, *args):
-    """Rebuild measurement upon unpickling using the application registry."""
-    return _unpickle(application_registry.Measurement, *args)
-
-
 def set_application_registry(registry):
     """Set the application registry, which is used for unpickling operations
     and when invoking pint.Quantity or pint.Unit directly.
@@ -127,6 +105,7 @@ __all__ = (
     "Unit",
     "UnitRegistry",
     "PintError",
+    "formatter",
     "DefinitionSyntaxError",
     "LogarithmicUnitCalculusError",
     "DimensionalityError",
diff --git a/pint/facets/measurement/objects.py b/pint/facets/measurement/objects.py
index ef70605..4a2ca4f 100644
--- a/pint/facets/measurement/objects.py
+++ b/pint/facets/measurement/objects.py
@@ -93,9 +93,9 @@ class Measurement(PlainQuantity):
 
     def __reduce__(self):
         # See notes in Quantity.__reduce__
-        from pint import _unpickle_measurement
+        from pint import _unpickle
 
-        return _unpickle_measurement, (Measurement, self.magnitude, self._units)
+        return _unpickle, (Measurement, self.magnitude, self._units)
 
     def __repr__(self):
         return f'Measurement({repr(self.magnitude.nominal_value)}, {repr(self.magnitude.std_dev)}, "{self.units}")'
diff --git a/pint/facets/plain/quantity.py b/pint/facets/plain/quantity.py
index d31bc5c..eefc3f1 100644
--- a/pint/facets/plain/quantity.py
+++ b/pint/facets/plain/quantity.py
@@ -152,12 +152,12 @@ class PlainQuantity[MagnitudeT: Magnitude](PrettyIPython, SharedRegistryObject):
         """Allow pickling quantities. Since UnitRegistries are not pickled, upon
         unpickling the new object is always attached to the application registry.
         """
-        from pint import _unpickle_quantity
+        from pint import _unpickle
 
         # Note: type(self) would be a mistake as subclasses built by
         # dinamically can't be pickled
         # TODO: Check if this is still the case.
-        return _unpickle_quantity, (PlainQuantity, self.magnitude, self._units)
+        return _unpickle, (PlainQuantity, self.magnitude, self._units)
 
     @overload
     def __new__(cls, value: MagnitudeT, units: UnitLike | None = None) -> Self: ...
diff --git a/pint/facets/plain/unit.py b/pint/facets/plain/unit.py
index 45dc065..f07c077 100644
--- a/pint/facets/plain/unit.py
+++ b/pint/facets/plain/unit.py
@@ -29,9 +29,9 @@ class PlainUnit(PrettyIPython, SharedRegistryObject):
 
     def __reduce__(self):
         # See notes in Quantity.__reduce__
-        from pint import _unpickle_unit
+        from pint import _unpickle
 
-        return _unpickle_unit, (PlainUnit, self._units)
+        return _unpickle, (PlainUnit, self._units)
 
     def __init__(self, units: UnitLike) -> None:
         super().__init__()
