diff --git a/.github/workflows/ci.yml b/.github/workflows/ci.yml
index 000d633..7e5329d 100644
--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -20,6 +20,8 @@ jobs:
         with:
           environments: typecheck
           activate-environment: true
+          version: PATH
+        continue-on-error: true
       - uses: jakebailey/pyright-action@v2
         continue-on-error: true
         with:
diff --git a/pyproject.toml b/pyproject.toml
index 42fd0f2..68b7d4b 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -128,8 +128,7 @@ lint = { features = ["lint"], no-default-feature = true }
 build = { features = ["build"], no-default-feature = true }
 test = { features = ["test"], solve-group = "default" }
 test-all = { features = ["test-all"], solve-group = "default" }
-typecheck = { features = ["typecheck", "test-all"], solve-group = "default" }
-numpy = { features = ["numpy"], solve-group = "default" }
+typecheck = { features = ["typecheck", "test-all", "py313"], solve-group = "default" }
 codspeed = { features = ["codspeed"], solve-group = "default" }
 docs = { features = [
   "docs",
@@ -223,7 +222,7 @@ include = ["pint"]
 exclude = ["pint/testsuite"]
 pythonVersion = "3.12"
 typeCheckingMode = "basic"
-
+strict = ["src/units/core.py"]
 [tool.pyright.defineConstant]
 HAS_BABEL = true
 HAS_UNCERTAINTIES = true
